backend/src/lib/report-generator.js

```diff
@@ -147,6 +190,121 @@ expressions.filters.changeID = function (input, prefix) {
+expressions.filters.d = function(input, s) {
+    return (input && input != "undefined") ? input : s;
+}
+
+// Display "From ... to ..." dates nicely, removing redundant information when the start and end date occur during the same month or year: {date_start | fromTo: date_end:'fr' | capfirst}
+// To internationalize or customize the resulting string, associate the desired output to the strings "from {0} to {1}" and "on {0}" in your Pwndoc translate file.
+expressions.filters.fromTo = function(start, end, locale) {
+    const start_date = new Date(start);
+    const end_date = new Date(end);
+    let options = {}, start_str = '', end_str = '';
+    let str = "from {0} to {1}";
+
+    if (start_date == "Invalid Date" || end_date == "Invalid Date") return start;
+
+    options = {day: '2-digit', month: '2-digit', year: 'numeric'};
+    end_str = end_date.toLocaleDateString(locale, options);
+
+    if (start_date.getYear() != end_date.getYear()) {
+        options = {day: '2-digit', month: '2-digit', year: 'numeric'};
+        start_str = start_date.toLocaleDateString(locale, options);
+    }
+    else if (start_date.getMonth() != end_date.getMonth()) {
+        options = {day: '2-digit', month: '2-digit'};
+        start_str = start_date.toLocaleDateString(locale, options);
+    }
+    else if (start_date.getDay() != end_date.getDay()) {
+        options = {day: '2-digit'};
+        start_str = start_date.toLocaleDateString(locale, options);
+    }
+    else {
+        start_str = end_str;
+        str = "on {0}";
+    }
+
+    return translate.translate(str).format(start_str, end_str);
+}
+
+// Group input elements by an attribute: {#findings | groupBy: 'severity'}{title}{/findings | groupBy: 'severity'}
+// Source: https://stackoverflow.com/a/34890276
+expressions.filters.groupBy = function(input, key) {
+    return input.reduce(function(rv, x) {
+        (rv[x[key]] = rv[x[key]] || []).push(x);
+        return rv;
+    }, {});
+}
+
+// Returns the initials from an input string (typically a firstname): {creator.firstname | initials}
+expressions.filters.initials = function(input) {
+    if (!input || input == "undefined") return input;
+    return input.replace(/(\w)\w+/gi,"$1.");
+}
+
+// Returns a string which is a concatenation of input elements using an optional separator string: {scope | join: ', '}
+// Can also be used to build raw OOXML strings.
+expressions.filters.join = function(input, sep = '') {
+    if (!input || input == "undefined") return input;
+    return input.join(sep);
+}
+
+// Returns the length (ie. number of items for an array) of input: {input | length}
+// Can be used as a conditional to check the emptiness of a list: {#input | length}Not empty{/input | length}
+expressions.filters.length = function(input) {
+    return input.length;
+}
+
+// Takes a multilines input strings (either raw or simple HTML paragraphs) and returns each line as an ordered list: {input | lines}
+expressions.filters.lines = function(input) {
```

Comment:
This is related to the issue #283, with the same use case: create a custom field where the auditors can fill a list of anything, for example the list of accounts used during the audit.

For now its indeed limited to WYSIWYG fields as multilines text input are not (yet) available as custom fields. WYSIWYG fields add a bit of clutter on the GUI but at least thank to this filter it gets the job done: auditors can enter the accounts one per line in the appropriate field, and these accounts are processed as a list in the template.

Depending on how #283 is implemented this filter might be redundant or not, but in doubt I thought it might still be useful to other users and provide a solution working right now instead of waiting for this issue to be resolved.

Moreover, I like the idea to keep the input field UI (WYSIWYG vs. multilines text fields) separated from the way the data is exposed to the template (text block vs. strings list), so people could actually use multilines text as an actual text block when they wish so, or use WYSIWYG fields to fill a list of elements if they need so (maybe a list screenshots / photographs, a list of reference links, etc.). In other words, I think it provides more freedom to Pwndoc users and therefore it seemed to me as something to share, but if you feel otherwise I can move it to my local filters.